plugins/vertexai/modelgauge/suts/vertexai_client.py

```diff
@@ -0,0 +1,95 @@
+import google.auth
+import httpx
+from google.auth.transport.requests import Request
+
+from modelgauge.secret_values import OptionalSecret, RequiredSecret, SecretDescription
+
+
+class VertexAIAPIKey(RequiredSecret):
+    """This key is used to request an access token from Google auth, which is then
+    used to call the inference endpoint. It is not used with the endpoint."""
+
+    @classmethod
+    def description(cls) -> SecretDescription:
+        return SecretDescription(
+            scope="vertexai",
+            key="api_key",
+            instructions="See https://cloud.google.com/docs/authentication/api-keys",
+        )
+
+
+class VertexAIProjectId(RequiredSecret):
+    @classmethod
+    def description(cls) -> SecretDescription:
+        return SecretDescription(
+            scope="vertexai",
+            key="project_id",
+            instructions="Your Google Cloud Platform project ID.",
+        )
+
+
+class VertexAIRegion(OptionalSecret):
+    @classmethod
+    def description(cls) -> SecretDescription:
+        return SecretDescription(
+            scope="vertexai",
+            key="region",
+            instructions="A Google Cloud Platform region.",
+        )
+
+
+class VertexAIClient:
+    def __init__(
+        self,
+        publisher: str,
+        model_name: str,
+        model_version: str,
+        streaming: bool,
+        api_key: VertexAIAPIKey,
+        project_id: VertexAIProjectId,
+        region: VertexAIRegion | str,
+    ):
+        self.publisher = publisher
+        self.model_name = model_name
+        self.model_version = model_version
+        self.api_key = api_key.value
+        self.project_id = project_id.value
+        self.streaming = streaming
+        if isinstance(region, str):
+            self.region = region
+        elif isinstance(region, VertexAIRegion):
+            self.region = region.value
+        else:
+            raise ValueError("Incorrect GCP region.")
+
+    def _get_access_token(self) -> str:
+        credentials, _ = google.auth.default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
```

Comment:
How does this work? Does it require you to have credentials setup in gcloud CLI?

Reply:
You don't need the gcloud CLI itself, but you do need the application default credentials. I'll add that to the documentation.